server/container/guice/cassandra-guice/src/test/java/org/apache/james/FixingGhostMailboxTest.java

```diff
@@ -0,0 +1,337 @@
+
+        fixGhostMailboxes(newAliceInbox);
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMessageList\", {\"filter\":{\"inMailboxes\":[\"" + newAliceInbox.getMailboxId().serialize() + "\"]}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("messageList"))
+            .body(ARGUMENTS + ".messageIds", hasSize(2))
+            .body(ARGUMENTS + ".messageIds", containsInAnyOrder(
+                message1.getMessageId().serialize(),
+                message2.getMessageId().serialize()));
+    }
+
+    @Test
+    public void webadminCanMergeTwoMailboxesRights() throws Exception {
+        Mailbox newAliceInbox = mailboxProbe.getMailbox(MailboxConstants.USER_NAMESPACE, alice, MailboxConstants.INBOX);
+        aclProbe.addRights(aliceInboxPath, cedric, MailboxACL.FULL_RIGHTS);
+
+        fixGhostMailboxes(newAliceInbox);
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMailboxes\", {\"ids\": [\"" + newAliceInbox.getMailboxId().serialize() + "\"]}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(FIRST_MAILBOX + ".sharedWith", hasKey(bob))
+            .body(FIRST_MAILBOX + ".sharedWith", hasKey(cedric));
+    }
+
+    @Test
+    public void oldGhostedMailboxShouldNoMoreBeAccessible() throws Exception {
+        Mailbox newAliceInbox = mailboxProbe.getMailbox(MailboxConstants.USER_NAMESPACE, alice, MailboxConstants.INBOX);
+        aclProbe.addRights(aliceInboxPath, cedric, MailboxACL.FULL_RIGHTS);
+
+        fixGhostMailboxes(newAliceInbox);
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMailboxes\", {\"ids\": [\"" + aliceGhostInboxId.serialize() + "\"]}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(ARGUMENTS + ".list", hasSize(0));
+    }
+
+    @Test
+    public void mergingMailboxTaskShouldBeInformative() {
+        Mailbox newAliceInbox = mailboxProbe.getMailbox(MailboxConstants.USER_NAMESPACE, alice, MailboxConstants.INBOX);
+
+        String taskId = fixGhostMailboxes(newAliceInbox);
+
+        given()
+            .port(webAdminProbe.getWebAdminPort())
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is(TaskManager.Status.COMPLETED.getValue()))
+            .body("taskId", is(taskId))
+            .body("additionalInformation.oldMailboxId", is(aliceGhostInboxId.serialize()))
+            .body("additionalInformation.newMailboxId", is(newAliceInbox.getMailboxId().serialize()))
```

Comment:
can we add more information data, such as the number of message which have been migrated, and if possible the total number of messages to migrate?

Reply:
I will get a look.